### PR TITLE
feat: display input description

### PR DESCRIPTION
**Pull request by jonasbrunvoll**
This PR includes a modal to display an optional description of the input. The modal is opened by clicking on the icon button. 

The modal have the following attributes
- [x] Focus on tab. 
- [x] Opens with enter.
- [x] Closes on esc, enter and clicking outside the modal body.
- [x] Arial-label to help the screen reader.   

Skjermbilde 2024-09-26 kl  16 44 37

Skjermbilde 2024-09-26 kl  16 44 48

Skjermbilde 2024-09-26 kl  16 45 02

### CODE REVIEW

**Comment by mortennordseth** [comment]
I think we need to have screen readers read the modal's content when it's opened, which it doesn't now. Also, when clicking outside the modal I guess it should close, right? The button should also have some sort of information that is read by the screen reader. When focusing on the button it says "button, main" now, and it's hard to understand the purpose of the button.

File: src/page-modules/contact/components/input/description-modal.tsx [hunk 1]
```diff
@@ -0,0 +1,62 @@
+import { Typo } from '@atb/components/typography';
+import style from './input.module.css';
+import { motion } from 'framer-motion';
+import { FocusScope } from '@react-aria/focus';
+import { MonoIcon } from '@atb/components/icon';
+import { useEffect } from 'react';
+
+type DescriptionModalProps = {
+  label: string;
```
Comment (mortennordseth) [inline src/page-modules/contact/components/input/description-modal.tsx:9]:
```suggestion
  title: string;
```

File: src/page-modules/contact/ticket-control/forms/postponePayment.tsx [hunk 2]
```diff
@@ -30,6 +30,7 @@ export const PostponePaymentForm = ({
           type="text"
           name="feeNumber"
           value={state.context.feeNumber || ''}
+          description={t(PageText.Contact.inputFields.feeNumber.description)}
```
Comment (mortennordseth) [inline src/page-modules/contact/ticket-control/forms/postponePayment.tsx:33]: You should also add it to the complaint form

**Comment by mortennordseth** [comment]
I think you should look into using `@react-aria` (https://react-spectrum.adobe.com/react-aria/useDialog.html) if we want to use modals 🤔

File: src/page-modules/contact/components/input/input.tsx [hunk 3]
```diff
@@ -18,31 +21,63 @@ export const Input = ({
...
   checked,
   value,
+  description,
   onChange,
 }: InputProps) => {
   const { t } = useTranslation();
+  const [isModalOpen, setIsModalOpen] = useState(false);
+  const displayDescriptionModal = description && isModalOpen; // Condition to avoid rendering of useEffect inside DescriptionModal before opened.
+
+  const handleModal = () => {
```
Comment (mortennordseth) [inline src/page-modules/contact/components/input/input.tsx:31]:
```suggestion
  const toggleModalState = () => {
```

File: src/page-modules/contact/components/input/input.tsx [hunk 4]
```diff
@@ -18,31 +21,63 @@ export const Input = ({
...
-      <label>
-        <Typo.span textType="body__primary">{t(label)}</Typo.span>
-      </label>
+      <div className={style.label_container}>
+        <label>
+          <Typo.span textType="body__primary">{t(label)}</Typo.span>
+        </label>
+
+        {description && (
+          <button
```
Comment (mortennordseth) [inline src/page-modules/contact/components/input/input.tsx:48]: Any reason to not use the Button component here?

File: src/translations/pages/contact.ts [hunk 5]
```diff
@@ -954,5 +954,21 @@ export const Contact = {
           ),
       },
     },
+    modal: {
+      open: {
+        ariaLabel: _(
+          'Klikk for å åpne modal',
```
Comment (mortennordseth) [inline src/translations/pages/contact.ts:960]: I think we should have a more descriptive label here. For example something like "More information about ${inputName}".

File: src/page-modules/contact/components/input/description-modal.tsx [hunk 6]
```diff
@@ -0,0 +1,94 @@
...
+      document.removeEventListener('keydown', handleEscape);
+      if (wrapperRef.current) {
+        wrapperRef.current.removeEventListener('click', handleClickOutside);
+      }
+    };
+  }, []);
+
+  if (!isModalOpen) return null;
+
+  return (
```
Comment (mortennordseth) [inline src/page-modules/contact/components/input/description-modal.tsx:57]: Would it be better to use [the dialog element](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/dialog) for this component? 🤔 At least add the dialog role to the element

Comment (mortennordseth) [reply src/page-modules/contact/components/input/description-modal.tsx:57]: I also think we need to add aria-labelledby and aria-describedby to help screen readers understand the content.

Comment (mortennordseth) [reply src/page-modules/contact/components/input/description-modal.tsx:57]: Probably aria-modal too: https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/Attributes/aria-modal

File: src/translations/pages/contact.ts (hunk 5 above)
Comment (jonasbrunvoll) [reply src/translations/pages/contact.ts:960]: Will update👍

File: src/page-modules/contact/components/input/input.tsx (hunk 4 above)
Comment (jonasbrunvoll) [reply src/page-modules/contact/components/input/input.tsx:48]: No, will update👍

File: src/page-modules/contact/components/input/description-modal.tsx [hunk 7]
```diff
@@ -0,0 +1,88 @@
...
+          ref={modalRef}
+          initial={{ opacity: 0, y: -30 }}
+          animate={{ opacity: 1, y: 0 }}
+          exit={{ opacity: 0, y: 0 }}
+          transition={{ ease: 'backOut', delay: 0.1, duration: 0.15 }}
+        >
+          <dialog
+            aria-modal="true"
+            aria-label={ariaLabel}
+            aria-labelledby={`modal-${title}`}
```
Comment (mortennordseth) [inline src/page-modules/contact/components/input/description-modal.tsx:69]: This should reference an element by ID (i.e. the h2 element inside the dialog in this case) (https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/Attributes/aria-labelledby)

File: src/translations/pages/contact.ts [hunk 8]
```diff
@@ -954,5 +954,20 @@ export const Contact = {
...
+      moreInformation: (inputName: string) =>
+        _(
+          `Mer informasjon om ${inputName}`,
+          `More information about ${inputName}`,
+          `Mer informasjon om ${inputName}`,
+        ),
+
+      close: (inputName: string) =>
+        _(
+          `Klikk for  å lukke informasjonsboks om  ${inputName}`,
```
Comment (mortennordseth) [inline src/translations/pages/contact.ts:967]:
```suggestion
          `Klikk for å lukke informasjonsboksen`,
```

File: src/translations/pages/contact.ts [hunk 9]
```diff
@@ -954,5 +954,20 @@ export const Contact = {
...
+          `Click to the information box ${inputName}`,
```
Comment (mortennordseth) [inline src/translations/pages/contact.ts:968]:
```suggestion
          `Click to close the information box`,
```

File: src/translations/pages/contact.ts [hunk 10]
```diff
@@ -954,5 +954,20 @@ export const Contact = {
...
+          `Klikk for å lukke informasjonsboksen  ${inputName}`,
```
Comment (mortennordseth) [inline src/translations/pages/contact.ts:969]:
```suggestion
          `Klikk for å lukke informasjonsboksen`,
```

**Review by mortennordseth — APPROVED** [review]

**Pull request merged**